include/bluetooth/bluetooth.h

```diff
@@ -772,9 +803,218 @@ int bt_le_ext_adv_get_info(const struct bt_le_ext_adv *adv,
  *  @param adv_type Type of advertising response from advertiser.
  *  @param buf Buffer containing advertiser data.
  */
+
 typedef void bt_le_scan_cb_t(const bt_addr_le_t *addr, int8_t rssi,
 			     uint8_t adv_type, struct net_buf_simple *buf);
 
+/**
+ * @brief Set or update the periodic advertisement parameters.
+ *
+ * The periodic advertisement parameters can only be set or updated on an
+ * extended advertisement set which is neither scannable, connectable or
+ * anonymous.
+ *
+ * @param adv	Advertising set object.
+ * @param param	Advertising parameters.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_set_param(struct bt_le_ext_adv *adv,
+			    const struct bt_le_per_adv_param *param);
+
+/**
+ * @brief Set or update the periodic advertisement data.
+ *
+ * The periodic advertisement data can only be set or updated on an
+ * extended advertisement set which is neither scannable, connectable or
+ * anonymous.
+ *
+ * @param adv		Advertising set object.
+ * @param data		Advertising data.
+ * @param data_len	Advertising data length.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_set_data(const struct bt_le_ext_adv *adv, const uint8_t *data,
+			   uint8_t data_len);
+
+/**
+ * @brief Enables or disables periodic advertisement.
+ *
+ * Enabling the periodic advertisement can be done independently of extended
+ * advertisement, but both periodic advertisement and extended advertisement
+ * shall be enabled before any data is sent.
+ * Enabling the periodic advertisement while it is already
+ * enabled updates the random address.
+ *
+ * @param adv		Advertising set object.
+ * @param enable	Whether to enable or disable periodic advertisement.
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_per_adv_enable(struct bt_le_ext_adv *adv, bool enable);
+
+struct bt_le_per_adv_sync_synced_info {
+	/** Advertiser LE address and type. */
+	const bt_addr_le_t *addr;
+
+	/** Periodic advertisement interval (N * 1.25MS) */
+	uint16_t interval;
+
+	/** Periodic advertisement advertiser clock accuracy (ppm) */
+	uint16_t clock_accuracy;
```

Comment:
We don't expose the clock_accuracy for connected callback (Connection Complete Event).
I would consider removing this parameter unless this is considered useful for the application.

Reply:
I'm not sure if it is or not, but I'll remove it :)